components/interactive-markdown.js

```diff
@@ -23,18 +28,49 @@ function ClickToRenderCodePreview(props) {
   )
 }
 
+function createCodeHandler(component) {
+  return function codeHandler(options, value) {
+    return {component, code: value, ...options}
+  }
+}
+
+function highlightWithPrism(code, language) {
+  return Prism.highlight(code, Prism.languages[language])
+}
+const prismHighlighter = highlighter(highlightWithPrism)
+
 function interactiveMarkdown(markdownString) {
   const componentBlocks = []
+  const highlightHtml = prismHighlighter('html')
+  const highlightJavaScript = prismHighlighter('javascript')
+  const highlightBash = prismHighlighter('bash')
+
+  const codeBlock = createCodeHandler(StaticCodeBlock)
+  const htmlCodeBlock = highlightHtml(codeBlock)
+  const javaScriptCodeBlock = highlightJavaScript(codeBlock)
+  const bashCodeBlock = highlightBash(codeBlock)
+  const codePreview = createCodeHandler(CodePreview)
+  const clickToRenderCodePreview = createCodeHandler(ClickToRenderCodePreview)
+
   const pragmaHandlers = {
     interactive(options, value) {
-      const component = options.clickToRender ?
-        ClickToRenderCodePreview :
-        CodePreview
-      return {component, code: value, ...options}
+      const interativeCodeBlock = options.clickToRender ?
+        clickToRenderCodePreview :
+        codePreview
+      return interativeCodeBlock(options, value)
     },
     callout(options, value) {
       return {component: Callout, children: value, ...options}
     },
+    js(options, value) {
```

Comment:
It looks like the current flow is "these are the pragmas we support (interactive, callout, js, bash, html, etc.)." If I use a new pragma in the markdown content, such as `css`, I do not know or expect to have to "enable" this pragma at this LOC. It seems ideal to take unknown pragmas and try to find a highlighter for it and in the case there truly is not one, then render a basic, non-highlighted code block.

The one caveat is that Prismjs (and I suspect many other syntax highlighting libraries) have non-core languages as separate CSS files/bundles. Even though we can process `css` for example, the styles to actually render it would still be missing. In either case, I think this would be an improvement, though; add styles for new pragma versus that PLUS registering it. Does that make sense/along the lines of what you were thinking?